src/rust/engine/fs/src/store.rs

```diff
@@ -684,7 +684,9 @@ impl Store {
     self
       .load_file_bytes_with(digest, move |bytes| {
         OpenOptions::new()
-          .create(true)
+          // Only create the file if it doesn't already exist; assume that if the file already
```

Comment:
What failed without this? ... it would be really, really awesome to fail for this case, and filter whatever we need to out of the `Snapshot`(s) to prevent the collisions in the first place.

Reply:
Independent runs of pants won't track materialisation state of classpath entries, they will make them fresh each time. We could try to write down materialisation state in the .digest files we create if we wanted maybe... But in general, this seems like a hard one to safely keep.